apis/apiextensions/v1/composition_transforms.go

```diff
@@ -275,6 +287,23 @@ func stringTrimTransform(input interface{}, t StringTransformType, trim string)
 	return str
 }
 
+func stringRegexpTransform(input interface{}, r string) (interface{}, error) {
+	if r == "" {
+		return nil, errors.New(errStringTransformTypeEmptyRegexp)
+	}
+
+	re, err := regexp.Compile(r)
+
+	if err != nil {
+		return nil, errors.Errorf(errStringTransformTypeRegexFailed, err.Error())
+	}
+
+	str := fmt.Sprintf("%v", input)
+	result := re.Find([]byte(str))
```

Comment:
I was just doing a final re-read before approving and it occurred to me that `re.Find` will either return `[]byte` or `nil`. I'm not sure how these values will be handled by the rest of the transform/patch code - in particular I suspect `[]byte` will be rendered as base64 encoded arbitrary bytes, not the expected value (presumably a string).

My suggestion here is:

* The transform should fail (i.e. return an error) if there is no match (i.e. `re.Find` returns `nil`).
* The transform should always type assert the returned `[]byte` into a ` string` before returning it.

Reply:
Given that I took so long to spot this I'm inclined to merge this and fix it in a follow-up PR. I'll do that once https://github.com/crossplane/crossplane/pull/3158 is merged. :)